admin-dev/themes/new-theme/js/components/router.js

```diff
@@ -43,8 +44,18 @@ const {$} = window;
  */
 export default class Router {
   constructor() {
+    EventEmitter.on('updateRoutes', (customRoutes) => {
```

Comment:
Question, why having a `beforeRoutes` even to to submit, and then the module must dispatch its own `updateRoutes` event to add its custom routes?

Its it easier to dispatch an event from the Router like `EventEmitter.emit('initRoutes', routeEvent);` which modules can register to, they would update the `routeEvent` to add their routes and then the Router can use them

It would avoid having two different events for one feature, and modules wouldn't need to emit anything they just listen to an event It feels more natural for modules to register to an event to extend a behaviour than dispatching their own

Reply:
To make it easier the `routeEvent` would have methods like `routeEvent.addRoutes()` to make additions more easily

Reply:
You'r right, gonna update it